src/utils/get-resume-state.ts

```diff
@@ -37,10 +37,11 @@ export function getResumeState(queryParams: string, storage: Storage): ResumeSta
     const storedSaksnummer = getStorageItem(StorageKey.SAKSNUMMER, storage);
 
     if (
-        storedKlageId !== null &&
-        queryTema === storedTema &&
-        queryYtelse === storedYtelse &&
-        querySaksnummer === storedSaksnummer
+        (storedKlageId !== null &&
+            queryTema === storedTema &&
+            queryYtelse === storedYtelse &&
+            querySaksnummer === storedSaksnummer) ||
+        (pathName !== '' && pathName === '/oppsummering' && storedKlageId !== null)
```

Comment:
Flytter svaret hit, siden det er samme prinsipp her.
Denne koden gjør det enda tydeligere også.

Om `pathName = "/oppsummering"` så er kan den umulig være en tom streng også. Derfor er det ikke noe poeng å teste begge deler.

Alt som testes her er `pathName === "/oppsummering" && storedKlageId !== null`.

Reply:
Ok, da er jeg med. Helt enig, den første sjekken er unødvendig nå. Tanken tidligere var å gjøre en null-sjekk, for å støtte den gamle bruken av funksjonen også etter endring av grensesnittet. Men det er unødvendig nå, som du påpeker.

Reply:
Når det er sagt, så mistenker jeg også at `storedKlageId !== null` ikke er nødvendig og at dette tilfellet kan flyttes oppover til:

```
    const storedKlageId = getStorageItem(StorageKey.KLAGE_ID, storage);

    if (pathName === '/oppsummering') {
        return {
            klageId: storedKlageId,
            saksnummer: null,
            tema: null,
            ytelse: null
        };
    }
```

Reply:
For å sette det i kontekst:
```ts
    const query = queryString.parse(queryParams);
    const queryKlageId = getQueryString(query, 'klageid');
    if (queryKlageId !== null) {
        return {
            klageId: queryKlageId,
            saksnummer: null,
            tema: null,
            ytelse: null
        };
    }

    const storedKlageId = getStorageItem(StorageKey.KLAGE_ID, storage);
    const storedTema = getStorageItem(StorageKey.TEMA, storage);
    const storedYtelse = getStorageItem(StorageKey.YTELSE, storage);
    const storedSaksnummer = getStorageItem(StorageKey.SAKSNUMMER, storage);

    if (pathName === '/oppsummering') {
        return {
            klageId: storedKlageId,
            saksnummer: storedSaksnummer,
            tema: storedTema,
            ytelse: storedYtelse
        };
    }

    const queryTema = getQueryString(query, 'tema');
    const queryYtelse = getQueryString(query, 'ytelse') ?? getDefaultYtelse(queryTema);
    const querySaksnummer = getQueryString(query, 'saksnummer');

    if (
        storedKlageId !== null &&
        queryTema === storedTema &&
        queryYtelse === storedYtelse &&
        querySaksnummer === storedSaksnummer
    ) {
        return {
            klageId: storedKlageId,
            tema: storedTema,
            ytelse: storedYtelse,
            saksnummer: storedSaksnummer
        };
    }

    return {
        klageId: null,
        tema: ensureStringIsTema(queryTema),
        ytelse: queryYtelse,
        saksnummer: querySaksnummer
    };
```

Reply:
Ja, jeg er enig i at det er tilstrekkelig. Jeg flytter det ikke ut av disjunksjonen, siden begge tilfellene har samme returverdi.

Reply:
Ok, den er grei. Jeg har for vane å returnere så tidlig som mulig og gjøre så lite som mulig.
I dette tilfellet gjør vi noen operasjoner på `query` og setter returverdiene i noen variabler, som aldri vil bli brukt til noe nyttig:
```ts
const queryTema = getQueryString(query, 'tema');
const queryYtelse = getQueryString(query, 'ytelse') ?? getDefaultYtelse(queryTema);
const querySaksnummer = getQueryString(query, 'saksnummer');
```

Reply:
Ålreit prinsipp det, kan endre det litt seinere/i morgen.